cli/src/commit_templater.rs

```diff
@@ -1671,3 +1732,81 @@ fn builtin_tree_diff_methods<'repo>() -> CommitTemplateBuildMethodFnMap<'repo, T
     // TODO: add files() or map() to support custom summary-like formatting?
     map
 }
+
+#[derive(Debug)]
+pub struct CryptographicSignature {
+    commit: Commit,
+}
+
+impl CryptographicSignature {
+    fn new(commit: Commit) -> Self {
+        if !commit.is_signed() {
+            panic!("commit must be signed");
+        }
+
+        Self { commit }
+    }
+
+    fn verify(&self) -> SignResult<Verification> {
+        self.commit
+            .verification()
+            .transpose()
+            .expect("must have signature")
+    }
+
+    fn status(&self) -> SignResult<SigStatus> {
+        self.verify().map(|verification| verification.status)
+    }
+
+    fn key(&self) -> Option<String> {
+        self.verify().map_or(None, |verification| verification.key)
+    }
+
+    fn display(&self) -> Option<String> {
+        self.verify()
+            .map_or(None, |verification| verification.display)
+    }
+}
+
+pub fn builtin_cryptographic_signature_methods<'repo>(
+) -> CommitTemplateBuildMethodFnMap<'repo, CryptographicSignature> {
+    type L<'repo> = CommitTemplateLanguage<'repo>;
+    // Not using maplit::hashmap!{} or custom declarative macro here because
+    // code completion inside macro is quite restricted.
+    let mut map = CommitTemplateBuildMethodFnMap::<CryptographicSignature>::new();
+    map.insert(
+        "status",
+        |_language, _diagnostics, _build_ctx, self_property, function| {
+            function.expect_no_arguments()?;
+            let out_property = TemplateFunction::new(self_property, |sig| {
+                let status = match sig.status() {
+                    Ok(SigStatus::Good) => "good",
+                    Ok(SigStatus::Bad) => "bad",
+                    Ok(SigStatus::Unknown) => "unknown",
+                    Err(_) => "invalid",
```

Comment:
I am not sure if we want to steamroller every error into "invalid", or if we want something more fine-grained.
[`SignError` defines two variants](https://github.com/pylbrecht/jj/blob/e3aebadb8a86d5d9a6e44eb1d359f2dafe588390/lib/src/signing.rs#L124-L131):
```rust
/// An error type for the signing/verifying operations
#[derive(Debug, Error)]
pub enum SignError {
    /// The verification failed because the signature *format* was invalid.
    #[error("Invalid signature")]
    InvalidSignatureFormat,
    /// A generic error from the backend impl.
    #[error("Signing error")]
    Backend(#[source] Box<dyn std::error::Error + Send + Sync>),
}
```

Maybe we only want to convert `InvalidSignatureFormat` to "invalid" and bubble up the other?

Reply:
I'm pretty sure that was exactly the point of separating out the explicitly invalid variant

Reply:
If it rarely occurs, I would just propagate all errors. The way it was implemented would be just because we didn't have a mechanism to propagate error from template functions.

Reply:
> If it rarely occurs, I would just propagate all errors. The way it was implemented would be just because we didn't have a mechanism to propagate error from template functions.

That would mean removing `"invalid"` entirely, correct?